Consistently format decorate block error messages.

drwells opened the pull request:
This is the followup to the discussion in #2566.

bangerth commented on cmake/setup_custom_targets.cmake:
```diff
@@ -60,13 +60,19 @@ FOREACH(_component compat_files documentation examples parameter_gui)
-        && ${CMAKE_COMMAND} -E echo "**  Error: Could not ${_description_string} disabled component \"${_component}\"."
-        && ${CMAKE_COMMAND} -E echo "**  Please reconfigure with -DDEAL_II_COMPONENT_${_componentuppercase}=ON"
+        && ${CMAKE_COMMAND} -E echo "${_decorated_first_message}"
+        && ${CMAKE_COMMAND} -E echo "${_decorated_second_message}"
```
This looks nice, but it isn't particularly easy to read what's happening here. What do others think about the approach to do the formatting in a macro, as opposed to just doing it by hand once when code is written?

davydden replied:
i agree that readability for developers suffers. But this is the way to make **consistent output** as in this case `_description_string` and `_component` are variables. So whatever formatting we decide on, it would be hard to do it consistently by hand. On the positive side for users, with macros messages will be formatted consistently and nicely.

So I personally :+1: this macro, but let's see if others want to contribute to the discussion.

davydden commented on cmake/macros/macro_decorate_with_stars.cmake:
```diff
@@ -0,0 +1,49 @@
+  MATH(EXPR _unpadded_line_length "${_line_length} - 4")
+
+  IF(${_message_length} LESS ${_unpadded_line_length})
+    MATH(EXPR _left_pad_size "(${_unpadded_line_length} - ${_message_length} + 1)/2")
```
nice trick to make the two pad sizes work with both odd and even differences.

davydden commented on cmake/setup_custom_targets.cmake:
```diff
@@ -60,13 +60,19 @@ FOREACH(_component compat_files documentation examples parameter_gui)
       COMMAND
            ${CMAKE_COMMAND} -E echo ''
         && ${CMAKE_COMMAND} -E echo ''
         && ${CMAKE_COMMAND} -E echo '***************************************************************************'
```
for consistency i would wrap line-of-starts in the same macro with
```
IF(${_message_length} EQ 0)
```
so that there is only one place where the length is defined.

davydden replied:
Nevermind, it is probably an overkill.For ouput string with known length we do it manually anyway.

davydden commented on cmake/macros/macro_decorate_with_stars.cmake:
```diff
@@ -0,0 +1,49 @@
+MACRO(DECORATE_WITH_STARS _message _decorated_message)
+  STRING(LENGTH ${_message} _message_length)
+  SET(_line_length 75)
+  MATH(EXPR _unpadded_line_length "${_line_length} - 4")
```
maybe `-6` to have at least a single space between `**` on both sides?

davydden commented on cmake/setup_custom_targets.cmake:
```diff
@@ -81,8 +87,8 @@ IF(NOT DEAL_II_COMPONENT_PACKAGE)
       && ${CMAKE_COMMAND} -E echo '***************************************************************************'
-      && ${CMAKE_COMMAND} -E echo "**  Error: Could not generate binary package. The component is disabled."
-      && ${CMAKE_COMMAND} -E echo "**  Please reconfigure with -DDEAL_II_COMPONENT_PACKAGE=ON"
+      && ${CMAKE_COMMAND} -E echo '**  Error: Could not generate binary package. The component is disabled. **'
```
@bangerth : from this line i would say the suggested usage is manual formatting for strings without variables and macro for strings with variables.

drwells commented:
Thank you for the review. Since I will force push in a moment I will just put all my comments here:
1. I wrote the macro so that lines whose length varies inside the loop could also be decorated without having to hardcode four distinct cases. In the other cases, where the length does not depend on local variables, I just hardcoded the stars. I agree that readability suffers slightly as a result. I will use better variable names to improve this.
2. I will switch to `-6` so that we always get spaces around stars. This won't effect the output but may make future output slightly nicer.

drwells commented:
I believe I addressed all of the review comments. I did leave the macro in so that those lines can be programmatically decorated; I do not know of a way around that. Is there anything else anyone would like to see in this PR?

davydden commented:
since there are no further comments, i am merging it.
